### ISSUE TITLE
[Accessibility]: Announce updates on inputs that control live page state

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35191]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518, #33512

### Problem description

<img width="884" alt="typeahead live" src="https://user-images.githubusercontent.com/103087/167504845-e25e381c-f1a4-48e4-9e80-c4ce967e487b.png">

Typing in this pattern input filters the number of results that are matched. It's difficult to determine if the user's input is actually filtering correctly or if there's a typo that matches nothing without navigating the page a significant amount.

<img width="884" alt="typeahead" src="https://user-images.githubusercontent.com/103087/167507611-add4a6e4-5a2f-409b-ad99-951c94b362db.png">

This is also true on the site-admin repo filters available on code intelligence policy edit pages.

### Expected behavior

These should be announced as the user types, or at the very least we should have an auxiliary count attached to the form input as a hint or label.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@efritz Going to assign this one to GitStart, LMK if you disagree/have already worked on a fix!

@gitstart-sourcegraph I think giving this `aria-live="polite"` would be a sufficient change to notify users
